### PR TITLE
Add model zoo input checkers

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

File: pkg/modelzooserver/input_check.go [hunk 1]
```diff
@@ -13,12 +13,39 @@
...
+	}
+	return nil
 }
 
 // checkImageURL checks a Docker image URL is valid
-func checkImageURL(imageURL string) bool {
-	return true
+func checkImageURL(imageURL string) error {
+	if len(imageURL) < 6 {
+		return fmt.Errorf("imageURL should have at least 6 characters")
```
Comment (wangkuiyi) [inline pkg/modelzooserver/input_check.go:39]: Why 6 characters?

File: pkg/modelzooserver/modelzooserver.go [hunk 2]
```diff
@@ -176,6 +176,10 @@ func (s *modelZooServer) ReleaseModelDef(stream pb.ModelZooServer_ReleaseModelDe
 	}
 	// close and flush the tar.gz file
 	fd.Close()
+	err = checkImageURL(reqName)
```
Comment (wangkuiyi) [inline pkg/modelzooserver/modelzooserver.go]: How about keep the consistent style as the following lines and write

```golang
if err := checkImageURL(..); err {
    return err
}
```

File: pkg/modelzooserver/input_check.go (hunk 1 above)
Comment (typhoonzero) [reply pkg/modelzooserver/input_check.go:39]: A model definition image name like `a` seems not meaningful.

File: pkg/modelzooserver/input_check.go [hunk 3]
```diff
@@ -13,12 +13,39 @@
...
 
+import (
+	"fmt"
+	"regexp"
+)
+
 // checkName checks model definition and trained model is valid.
-func checkName(name string) bool {
-	return true
+func checkName(name string) error {
```
Comment (weiguoz) [inline pkg/modelzooserver/input_check.go:22]: How about adding a few unit tests as samples?

Comment (typhoonzero) [reply pkg/modelzooserver/input_check.go:22]: Thanks, I just forgot to `git add` the test file...

**Review by weiguoz — APPROVED** [review]

**Pull request merged**